scanner/src/main/kotlin/provenance/ProvenanceDownloader.kt

```diff
@@ -113,8 +116,12 @@ class DefaultProvenanceDownloader(
             }
 
             // We need to make a copy of the working tree, because it could be used by another coroutine after this
-            // call has finished.
-            root.toPath().copyToRecursively(downloadDir.toPath(), followLinks = false, overwrite = true)
+            // call has finished. Tolerate if copying lockfiles fails.
+            val onError = { source: Path, _: Path, _: Exception ->
+                if (source.extension == "lock") OnErrorResult.SKIP_SUBTREE else OnErrorResult.TERMINATE
+            }
+
+            root.toPath().copyToRecursively(downloadDir.toPath(), onError, followLinks = false, overwrite = true)
```

Comment:
@mnonnenmacher, maybe the better solution is to use the [`git worktree` command](https://git-scm.com/docs/git-worktree) here instead, to create a "linked" worktree from the "main" worktree?

Reply:
Yes, that could be a good usecase for `git worktree`.